### PR TITLE
PP-7583 Url Structure > Add CSRF middleware to account router

### PR DESCRIPTION
**Pull request by iqbalgds**
- Add the CSRF middleware function 'validateAndRefreshCsrf' to the account router.

### CODE REVIEW

File: app/routes.js [hunk 1]
```diff
@@ -102,7 +102,7 @@ module.exports.paths = paths
 
 module.exports.bind = function (app) {
   const account = new Router({ mergeParams: true })
-  account.use(getServiceAndAccount, userIsAuthorised)
+  account.use(getServiceAndAccount, userIsAuthorised, validateAndRefreshCsrf)
```
Comment (kbottla) [inline app/routes.js:105]: Looks like we need `ensureSessionHasCsrfSecret` where creates new csrfSecret if one doesn't exist on session.

```suggestion
  account.use(getServiceAndAccount, userIsAuthorised, ensureSessionHasCsrfSecret, validateAndRefreshCsrf)
```

**Review by kbottla — CHANGES_REQUESTED** [review]

File: app/routes.js (hunk 1 above)
Comment (iqbalgds) [reply app/routes.js]: Just updated it - thanks.

**Review by kbottla — APPROVED** [review]
LGTM!

File: app/routes.js [hunk 2]
```diff
@@ -102,7 +102,7 @@ module.exports.paths = paths
 
 module.exports.bind = function (app) {
   const account = new Router({ mergeParams: true })
-  account.use(getServiceAndAccount, userIsAuthorised)
+  account.use(getServiceAndAccount, userIsAuthorised, ensureSessionHasCsrfSecret, validateAndRefreshCsrf)
```
Comment (sfount) [inline app/routes.js:105]: Apologies I missed this before it was reviewed, I don't think you need `ensureSessionHasCsrfSecret` here, I believe that's done once when the session is initially authorised and created and should then be subsequently available -- I think the issue with the tests passing was with the payment types integration test

**Pull request merged**